src/ports/postgres/modules/elastic_net/elastic_net_generate_result.py_in

```diff
@@ -13,50 +15,131 @@ def __elastic_net_generate_result (optimizer, iteration_run, **args):
+        out_table_qstr = """
+        SELECT
+            {grouping_str1}
+            (result).coefficients AS coef,
+            (result).intercept AS intercept
+        FROM (
+            SELECT {schema_madlib}.{result_func} AS result,
+            {col_grp_key}
+            FROM {tbl_state}
+            WHERE {col_grp_iteration} = {iteration_run}
         ) t
-        """.format(result_func = result_func, tbl_state = tbl_state).format(
+        JOIN
+        (
+            SELECT
+                {grouping_str1}
+                array_to_string(ARRAY[{grouping_str}],
+                                ','
+                               ) AS {col_grp_key}
+            FROM {source_table}
+                {groupby_str}
+        ) n_tuples_including_nulls_subq
+        {using_str}
+        """.format(result_func = result_func, tbl_state = tbl_state,
+            col_grp_iteration = args["col_grp_iteration"],
+            iteration_run = iteration_run,
+            groupby_str=groupby_str,
+            grouping_str1=grouping_str1,
+            grouping_str=grouping_str,
+            using_str=using_str,
+            col_grp_key=col_grp_key,
+            source_table=source_table,
+            schema_madlib = args["schema_madlib"])
+    else:
+        ## Its a much simpler query when there is no grouping.
+        select_grouping_info = ""
+        out_table_qstr = """
+            SELECT
+                (result).coefficients AS coef,
+                (result).intercept AS intercept
+            FROM (
+                SELECT {schema_madlib}.{result_func} AS result
+                FROM {tbl_state}
+                WHERE {col_grp_iteration} = {iteration_run}
+            ) t
+        """.format(result_func = result_func, tbl_state = tbl_state,
+            col_grp_iteration = args["col_grp_iteration"],
             iteration_run = iteration_run,
-            **args))[0]
+            schema_madlib = args["schema_madlib"])
 
-    r_coef = mad_vec(result["coef"], text = False)
+    ## Create the output table
+    plpy.execute("""
+             DROP TABLE IF EXISTS {tbl_result};
+             CREATE TABLE {tbl_result} (
+                 {select_grouping_info}
+                 family            text,
+                 features          text[],
+                 features_selected text[],
+                 coef_nonzero      double precision[],
+                 coef_all          double precision[],
+                 intercept         double precision,
+                 log_likelihood    double precision,
+                 standardize       boolean,
+                 iteration_run     integer)
+             """.format(select_grouping_info=select_grouping_info, **args))
+
+    result = plpy.execute(out_table_qstr)
+    for res in result:
+        build_output_table(res, grouping_column, grouping_str1,
+            standardize_flag, iteration_run, **args)
 
+    ## Create summary table, listing the grouping columns used.
```

Comment:
These were the statistics that were already present in elastic net. Adding more stats to the summary table might be possible, but would that warrant a broader discussion?